services/sonar/sonar-quality-gate.service.js

```diff
@@ -0,0 +1,47 @@
+'use strict'
+
+const SonarBase = require('./sonar-base')
+const { patternBase, queryParamSchema } = require('./sonar-helpers')
+
+module.exports = class SonarQualityGate extends SonarBase {
+  static get category() {
+    return 'analysis'
+  }
+
+  static get defaultBadgeData() {
+    return { label: 'quality gate' }
+  }
+
+  static render({ qualityState }) {
+    if (qualityState === 'OK') {
+      return {
+        message: 'passed',
```

Comment:
I almost used our build status messages (`passing` and `failing`) here, but decided to go with the wording Sonar uses for the quality gate metric since this isn't a build status badge